reconcile/utils/pagerduty_api.py

```diff
@@ -96,4 +137,77 @@ def get(self, name):
         Returns:
             PagerDutyApi: PagerDutyApi instance
         """
-        return self.pd_apis.get(name)
+        return self.pd_apis[name]
+
+
+def get_pagerduty_map(
+    secret_reader: SecretReader,
+    pagerduty_instances: Iterable[SupportsPagerDutyInstance],
+    init_users: bool = True,
+    pager_duty_api_class: type[PagerDutyApi] = PagerDutyApi,
+) -> PagerDutyMap:
+    """Initiate a PagerDutyMap for given PagerDuty instances."""
+    return PagerDutyMap(
+        instances=[
+            PagerDutyConfig(name=i.name, token=secret_reader.read_secret(i.token))
+            for i in pagerduty_instances
+        ],
+        init_users=init_users,
+        pager_duty_api_class=pager_duty_api_class,
+    )
+
+
+def get_pagerduty_name(user: SupportsUser) -> str:
+    return user.pagerduty_username or user.org_username
+
+
+@retry(no_retry_exceptions=PagerDutyTargetException)
+def get_usernames_from_pagerduty(
```

Comment:
this is the replacement for the one in  `slack_usergroups.py`, right?
are you going to remove the other one?

Reply:
Yes, this PR is a preparation step for the `slack-usergroups` refactoring